fwsrc/user/user_main.c

```diff
@@ -80,35 +222,48 @@ void user_init(void)
 		while(1) { uart0_sendStr( "\r\nFAULT\r\n" ); }
 	}
 
+	// Initialize common settings
 	CSInit();
 
+	// Start MDNS services
 	SetServiceName( "espcom" );
 	AddMDNSName(    "esp82xx" );
 	AddMDNSName(    "espcom" );
 	AddMDNSService( "_http._tcp",    "An ESP82XX Webserver", WEB_PORT );
 	AddMDNSService( "_espcom._udp",  "ESP82XX Comunication", COM_PORT );
 	AddMDNSService( "_esp82xx._udp", "ESP82XX Backend",      BACKEND_PORT );
 
-	//Add a process
-	system_os_task(procTask, procTaskPrio, procTaskQueue, procTaskQueueLen);
-
-	//Timer example
+	// Set timer100ms to be called every 100ms
 	os_timer_disarm(&some_timer);
-	os_timer_setfn(&some_timer, (os_timer_func_t *)myTimer, NULL);
+	os_timer_setfn(&some_timer, (os_timer_func_t *)timer100ms, NULL);
 	os_timer_arm(&some_timer, 100, 1);
 
-	printf( "Boot Ok.\n" );
+	os_printf( "Boot Ok.\n" );
 
+	// Set the wifi sleep type
 	wifi_set_sleep_type(LIGHT_SLEEP_T);
 	wifi_fpm_set_sleep_type(LIGHT_SLEEP_T);
 
+	// Add a process and start it
+	system_os_task(procTask, procTaskPrio, procTaskQueue, procTaskQueueLen);
 	system_os_post(procTaskPrio, 0, 0 );
 }
 
+/**
+ * This will be called to disable any interrupts should the firmware enter a
+ * section with critical timing. There is no code in this project that will
+ * cause reboots if interrupts are disabled.
+ */
+void ICACHE_FLASH_ATTR EnterCritical(void)
```

Comment:
These almost certainly cannot have the ICACHE_FLASH_ATTR set on them, otherwise you will need to exit to main system flash instead of IRAM from a critical location.  This may in very rare cases cause a serious issue.

Reply:
Fair point. I'll have to change this in swadge2019 as well. FWIW, it hasn't been causing issues there, but the rare case may not have been hit yet.

Reply:
Fixed